### PR TITLE
Update on paste handler

### PR DESCRIPTION
**Pull request by aljordan82**

### CODE REVIEW

File: source/Clipboard.js [hunk 1]
```diff
@@ -83,7 +83,26 @@ var onPaste = function ( event ) {
     if ( items && ( hasFiles || ( !isEdge && !hasHtml ))) {
         event.preventDefault();
         l = items.length;
-        while ( l-- ) {
+
+        // Trigger a willPaste event if these is an image type on the clipboardData.
```
Comment (akauffmanGG) [inline source/Clipboard.js]: Trigger a willPaste event if _there_ is an image type on the clipboardData

File: source/Clipboard.js [hunk 2]
```diff
@@ -83,7 +83,26 @@ var onPaste = function ( event ) {
...
+        for (var i = items.length - 1; i >= 0; i--) {
```
Comment (akauffmanGG) [inline source/Clipboard.js]: Needs to match spacing standards.

File: source/Clipboard.js [hunk 3]
```diff
@@ -83,7 +83,26 @@ var onPaste = function ( event ) {
...
+                var imagePasteEvent = {
+                    clipboardData: event.clipboardData,
+                    isImage: true,
+                    preventDefault: function () {
+                        this.defaultPrevented = true;
+                    },
+                    defaultPrevented: false
+                };
+
+                this.fireEvent( 'willPaste', imagePasteEvent);
```
Comment (akauffmanGG) [inline source/Clipboard.js]: What if there is more than one image on the clipboard? Do we want to fire multiple paste events?

File: source/Clipboard.js [hunk 4]
```diff
@@ -83,7 +83,26 @@ var onPaste = function ( event ) {
     if ( items && ( hasFiles || ( !isEdge && !hasHtml ))) {
         event.preventDefault();
         l = items.length;
-        while ( l-- ) {
+
+        // Trigger a willPaste event if these is an image type on the clipboardData.
+        for (var i = items.length - 1; i >= 0; i--) {
+            if ( /^image\/.*/.test( items[i].type ) ) {
+                hasImage = true;
```
Comment (akauffmanGG) [inline source/Clipboard.js]: hasImage doesn't appear to be used anywhere

File: source/Clipboard.js (hunk 3 above)
Comment (aljordan82) [inline source/Clipboard.js]: The willPaste event has clipboard data, and it is up to the consumer to handle the paste data.
also,

**Pull request merged**